### ISSUE TITLE
`no-unused-vars` bug

### ISSUE THREAD

**Issue by hacdias** [eslint/eslint#12458]
**Tell us about your environment**

* **ESLint Version:** v6.5.1
* **Node Version:** v12.8.0
* **npm Version:** v6.11.3

**What parser (default, Babel-ESLint, etc.) are you using?**

ESLint with Babel and presets 'react-app' and 'standard'.

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: "babel-eslint",
  extends: ['react-app', 'standard']
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
    for (const { path, hash } of res) {
      // Only go for direct children
      if (path.indexOf('/') === -1 && path !== '') {
        const src = `/ipfs/${hash}`
        const dst = join(realMfsPath(root), path)

        try {
          await ipfs.files.cp([src, dst])
        } catch (err) {
          throw Object.assign(new Error('Folder already exists.'), { code: 'ERR_FOLDER_EXISTS' })
        }
      }
    }
```

```bash
eslint
```

**What did you expect to happen?**

No errors/warnings.

**What actually happened? Please include the actual, raw output from ESLint.**

```
/.../actions.js
  209:18  error  'path' is defined but never used  no-unused-vars
  209:24  error  'hash' is defined but never used  no-unused-vars
```

The same error happened in more places. This is flagging those deconstructed variables in the `for` as unused. However, if I remove them, it the will say they are not defined.

This is preventing me from updating to the newers `react-scripts` and I believe it might be a bug here.

**Are you willing to submit a pull request to fix this bug?**

Not aware of the ESLint internals.

**Comment by g-plane** [comment]
Thanks for this issue.

Can you upgrade your "babel-eslint" to latest version and try again? Thanks.

**Comment by hacdias** [comment]
@g-plane solved it! Thank you very much.
